core/backend/src/test/imodel/IModel.test.ts

```diff
@@ -3303,4 +3303,67 @@ describe("iModel", () => {
     }
     testImodel.close();
   });
+
+  it("should provide meaningful error when querying a closed iModel", async () => {
+    const testImodel = SnapshotDb.createEmpty(IModelTestUtils.prepareOutputFile("IModel", "QueryingClosedImodel.bim"), { rootSubject: { name: "QueryClosedTest" } });
+    assert.isTrue(testImodel.isOpen);
+
+    // Close the iModel for the tests
+    testImodel.close();
+    assert.isFalse(testImodel.isOpen);
+
+    const queryOperations = [
+      {
+        name: "withPreparedSqliteStatement",
+        operation: () => testImodel.withPreparedSqliteStatement("SELECT 1", () => { }),
+        expectedErrorMessage: "Cannot query a closed Db"
+      },
+      {
+        name: "prepareSqliteStatement",
+        operation: () => testImodel.prepareSqliteStatement("SELECT 1"),
+        expectedErrorMessage: "Cannot query a closed Db"
+      },
+      {
+        name: "prepareStatement",
+        operation: () => testImodel.prepareStatement("SELECT 1"), // eslint-disable-line @typescript-eslint/no-deprecated
+        expectedErrorMessage: "Cannot query a closed Db"
+      },
+      {
+        name: "withPreparedStatement",
+        operation: () => testImodel.withPreparedStatement("SELECT ECInstanceId FROM BisCore:Element LIMIT 1", () => { }), // eslint-disable-line @typescript-eslint/no-deprecated
+        expectedErrorMessage: "Cannot query a closed Db"
+      },
+      {
+        name: "elements.queryChildren",
+        operation: () => testImodel.elements.queryChildren(IModel.rootSubjectId),
+        expectedErrorMessage: "Cannot query a closed Db"
+      },
+      {
+        name: "elements.getAspects",
+        operation: () => testImodel.elements.getAspects("0x1", "WrongSchema:WrongClass"),
+        expectedErrorMessage: "db is not open"
+      }
+    ];
+
+    for (const testCase of queryOperations) {
+      try {
+        testCase.operation();
+        assert.fail(`${testCase.name} should have thrown an error`);
+      } catch (error: any) {
+        // Verify error message
+        assert.include(error.message, testCase.expectedErrorMessage,
+          `${testCase.name} should contain "${testCase.expectedErrorMessage}" in message, got "${error.message}"`);
+      }
+    }
```

Comment:
```suggestion
    const closedDbError = "Cannot query a closed Db";
    expect(() => testImodel.withPreparedSqliteStatement("SELECT 1", () => { })).to.throw(closedDbError);
    expect(() => testImodel.withPreparedSqliteStatement("SELECT 1", () => { })).to.throw(closedDbError);
    expect(() => testImodel.prepareSqliteStatement("SELECT 1")).to.throw(closedDbError);
    // eslint-disable-next-line @typescript-eslint/no-deprecated
    expect(() => testImodel.prepareStatement("SELECT 1")).to.throw.closedDbError);
    // eslint-disable-next-line @typescript-eslint/no-deprecated
    expect(() => testImodel.withPreparedStatement("SELECT ECInstanceId FROM BisCore:Element LIMIT 1", () => { })).to.throw(closedDbError);
    expect(() => testImodel.elements.queryChildren(IModel.rootSubjectId)).to.throw(closedDbError);
    expectedErrorMessage: "Cannot query a closed Db"
    expect(() => testImodel.elements.getAspects("0x1", "WrongSchema:WrongClass")).to.throw("db is not open");

```

Way simpler.
BTW why should `getAspects` produce a different "not open" error than all the other queries?

Reply:
Done!
Yeah, getAspects hits a different closed db check in native before the query execution code.
Also, createQueryReader has a check on the ts side itself.
I thought it best to include both as well.